src/Microsoft.PowerShell.Commands.Utility/commands/utility/UtilityCommon.cs

```diff
@@ -206,6 +206,58 @@ private set
         /// </summary>
         public string Path { get; private set; }
 
+        /// <summary>
+        /// Gets the hexadecimal representation of the <see cref="Offset64"/> value.
+        /// </summary>
+        public string HexOffset { get => string.Format("{0:X16}", Offset64); }
+
+        private const int BytesPerLine = 16;
+
+        /// <summary>
+        /// Gets a space-delimited string of the <see cref="Bytes"/> in this <see cref="ByteCollection"/>
+        /// in hexadecimal format.
+        /// </summary>
+        public string HexBytes
+        {
+            get
+            {
+                StringBuilder line = new StringBuilder(BytesPerLine * 3);
+
+                foreach (var currentByte in Bytes)
+                {
+                    line.AppendFormat("{0:X2} ", currentByte);
+                }
+
+                return line.ToString().Trim();
```

Comment:
Why do we need the Trim()? Can we do a right format without extra spaces?

Reply:
I don't see a simple way to do so without checking every iteration, which seems more expensive.

Reply:
I don't understand where do the extra spaces come from? Look like we use broken format.

Reply:
There's only one extra space at the end -- format sequence is `'{0:X2} '` to easily have spaces in between each byte for readability. Not sure if there's a more efficient way to trim that off?